homeassistant/components/shelly/coordinator.py

```diff
@@ -146,6 +153,43 @@ def async_setup(self) -> None:
         )
         self.device_id = device_entry.id
 
+    async def _async_device_connect(self) -> None:
+        """Connect to a Shelly Block device."""
+        LOGGER.debug("Connecting to Shelly Device - %s", self.name)
+        try:
+            await self.device.initialize()
+            update_device_fw_info(self.hass, self.device, self.entry)
+        except DeviceConnectionError as err:
+            raise UpdateFailed(f"Device disconnected: {repr(err)}") from err
+        except InvalidAuthError:
+            self.entry.async_start_reauth(self.hass)
+            return
+
+        if not self.device.firmware_supported:
+            async_create_issue_unsupported_firmware(self.hass, self.entry)
+            return
+
+        if not self.finish_setup_platforms:
+            return
+
+        LOGGER.debug("Device %s is online, resuming setup", self.entry.title)
+        platforms = self.finish_setup_platforms
+        self.finish_setup_platforms = None
+
+        data = {**self.entry.data}
+
+        # Update sleep_period
+        if isinstance(self.device, RpcDevice):
+            data[CONF_SLEEP_PERIOD] = get_rpc_device_wakeup_period(self.device.status)
+        elif isinstance(self.device, BlockDevice):
+            data[CONF_SLEEP_PERIOD] = get_block_device_sleep_period(
+                self.device.settings
+            )
+        self.hass.config_entries.async_update_entry(self.entry, data=data)
```

Comment:
How frequently is this going to change? Should we only call this if the data has changed?